x-pack/plugins/fleet/server/types/rest_spec/settings.ts

```diff
@@ -1,40 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-import { schema } from '@kbn/config-schema';
-
-import { isDiffPathProtocol } from '../../../common/services';
-
-export const GetSettingsRequestSchema = {};
-
-export const PutSettingsRequestSchema = {
-  body: schema.object({
-    fleet_server_hosts: schema.maybe(
-      schema.arrayOf(schema.uri({ scheme: ['http', 'https'] }), {
-        validate: (value) => {
-          if (value.length && isDiffPathProtocol(value)) {
-            return 'Protocol and path must be the same for each URL';
```

Comment:
the validation is not exactly the same in openapi, we should probably add api tests to cover the validation
```
                fleet_server_hosts:
                  type: array
                  description: Protocol and path must be the same for each URL
                  items:
                    type: string
                    format: uri
```